src/components/Card.js

```diff
@@ -35,18 +51,29 @@ export default class Card {
     //like button
     this._likeButton.addEventListener("click", () => this._handleLikeBtn());
 
-    this._trashButton.addEventListener("click", () => this._handleDelete());
+    this._trashButton.addEventListener("click", () =>
+      this._handleDeleteClick(this)
+    );
 
     this._cardImageEl.addEventListener("click", () =>
       this._handleImageClick(this._name, this._link)
     );
   }
 
   _handleLikeBtn() {
+    this._isLiked = !this._isLiked;
```

Comment:
`    this._isLiked = !this._isLiked;
`

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.


Please, move it into `toggleLikeBtn`